cmd/cat/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+func checkFile(e error) {
+	if e != nil {
+		log.Fatal("Incorrect file path or file does not exist ", e)
+	}
+}
+
+func main() {
+	var n bool
+	flag.BoolVar(&n, "n", false, "number of lines")
+	flag.Parse()
+	if len(os.Args) == 0 {
```

Comment:
os.args[0] is always the executable you are running, If you need to make sure the user passed an argument you need to check if the length is < 2